advanced-achievements-plugin/src/main/java/com/hm/achievement/db/AchievementCache.java

```diff
@@ -0,0 +1,89 @@
+package com.hm.achievement.db;
+
+import com.hm.achievement.achievement.Achievement;
+import com.hm.achievement.achievement.AchievementBuilder;
+import com.hm.achievement.category.MultipleAchievements;
+import com.hm.achievement.category.NormalAchievements;
+import com.hm.achievement.exception.PluginLoadError;
+import com.hm.achievement.lifecycle.Reloadable;
+import com.hm.mcshared.file.CommentedYamlConfiguration;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class AchievementCache implements Reloadable {
+
+	private final Set<Achievement> cache = new HashSet<>();
+	private final Map<String, Achievement> nameMap = new HashMap<>();
+	private final Map<String, Set<Achievement>> categoryMap = new HashMap<>();
+	private final CommentedYamlConfiguration mainConfig;
+
+	@Inject
+	public AchievementCache(@Named("main") CommentedYamlConfiguration mainConfig) {
+		this.mainConfig = mainConfig;
+	}
+
+	public Achievement getByName(String name) {
+		return nameMap.get(name.toLowerCase());
+	}
+
+	public Set<Achievement> getByCategory(String category) {
+		return categoryMap.getOrDefault(category.toLowerCase(), new HashSet<>());
+	}
+
+	public void load() {
+		cache.clear();
+		nameMap.clear();
+		for (NormalAchievements value : NormalAchievements.values()) {
+			if (mainConfig.contains(value.getCategoryName())) {
```

Comment:
To ensure that you havnt commented out that category, I added the check to the MultipleAchievements too

Reply:
Actually the code in _FileUpdater_ will reinstate the category as empty if it's commented out, so that check is superfluous. I suggest either:
* removing it completely.
* changing it to `if (disabledCategories.contains(value))` (you can inject `disabledCategories` very easily using Dagger).